cmd/init.go

```diff
@@ -1,178 +1,119 @@
-// a couple of flags.
-type initConfig struct {
-	// Verbose mode instructs the system to output detailed logs as the command
-	// progresses.
-	Verbose bool
+func runInit(cmd *cobra.Command, args []string) error {
+	config := newInitConfig(args).Prompt()
+
+	function := faas.Function{
+		Name:    config.Name,
+		Root:    config.Path,
+		Runtime: config.Runtime,
+		Trigger: config.Trigger,
+	}
 
+	client := faas.New(
+		faas.WithVerbose(config.Verbose),
+		faas.WithTemplates(config.Templates))
+
+	return client.Initialize(function)
+}
+
+type initConfig struct {
 	// Name of the service in DNS-compatible format (ex myfunc.example.com)
 	Name string
 
-	// Trigger is the form of the resultant function, i.e. the function signature
-	// and contextually avaialable resources.  For example 'http' for a funciton
-	// expected to be invoked via straight HTTP requests, or 'events' for a
-	// function which will be invoked with CloudEvents.
-	Trigger string
+	// Path to files on disk.  Defaults to current working directory.
+	Path string
+
+	// Runtime language/framework.
+	Runtime string
 
 	// Templates is an optional path that, if it exists, will be used as a source
 	// for additional templates not included in the binary.  If not provided
 	// explicitly as a flag (--templates) or env (FAAS_TEMPLATES), the default
 	// location is $XDG_CONFIG_HOME/templates ($HOME/.config/faas/templates)
 	Templates string
 
-	// Runtime is the first argument, and specifies the resultant Function
-	// implementation runtime.
-	Runtime string
+	// Trigger is the form of the resultant Function, i.e. the Function signature
+	// and contextually avaialable resources.  For example 'http' for a funciton
+	// expected to be invoked via straight HTTP requests, or 'events' for a
+	// Function which will be invoked with CloudEvents.
+	Trigger string
 
-	// Path of the Function implementation on local disk. Defaults to current
-	// working directory of the process.
-	Path string
+	// Verbose logging enabled.
+	Verbose bool
 
-	// Image tag for the resulting Function
-	Tag string
+	// Yes: agree to values arrived upon from environment plus flags plus defaults,
+	// and skip the interactive prompting (only applicable when attached to a TTY).
+	Yes bool
 }
 
-func initializeProject(cmd *cobra.Command, args []string) (err error) {
-	if len(args) == 0 {
-		return errors.New("'faas init' requires a runtime argument")
+// newInitConfig returns a config populated from the current execution context
+// (args, flags and environment variables)
+func newInitConfig(args []string) initConfig {
+	var name string
+	if len(args) > 0 {
+		name = args[0] // If explicitly provided, use.
 	}
-
-	var config = initConfig{
-		Runtime:   args[0],
-		Verbose:   viper.GetBool("verbose"),
-		Name:      viper.GetString("name"),
+	return initConfig{
+		Name:      deriveName(name, viper.GetString("path")), // args[0] or derived
 		Path:      viper.GetString("path"),
-		Trigger:   viper.GetString("trigger"),
```

Comment:
Why remove this from the config if it's offered as a CLI option?

Reply:
I believe my thought was that the value of 'trigger' is a transient value used only during initialization, is not used once the Function has been initialized (prior to there being a config at all), and so it is not included.  However, the name of the template used when originally initializing the function could be useful information.  So that is now part of the serialized Function config again.